rpc.go

```diff
@@ -475,14 +488,27 @@ loop:
 	// the ResultChan for the remaining RPCs. If not ready the result
 	// will be the context error.
 	for _, rpc := range rpcs[canceledIndex:] {
+		description := rpc.Description()
 		select {
 		case res := <-rpc.ResultChan():
+			duration := time.Since(rpcStartTime).Seconds()
 			results[rpcToRes[rpc]] = res
 			if res.Error != nil {
 				c.handleResultError(res.Error, rpc.Region(), rc)
+				// Track failed operation due to cancellation or other error
+				o := operationDurationSeconds.WithLabelValues(description, "error")
+				observability.ObserveWithTrace(ctx, o, duration)
+			} else {
+				// Track successful operation that completed after cancellation
+				o := operationDurationSeconds.WithLabelValues(description, "ok")
+				observability.ObserveWithTrace(ctx, o, duration)
 			}
 		default:
 			results[rpcToRes[rpc]].Error = ctx.Err()
+			// Track operation that was canceled
+			duration := time.Since(rpcStartTime).Seconds()
+			o := operationDurationSeconds.WithLabelValues(description, "error")
```

Comment:
In this case, I think "error" might be misleading to any SLI metrics; because in this case, we are not erroring because of HBase returning back a specific response, we are erroring because the context got cancelled.